m4opt/missions/_lsst.py

```diff
@@ -0,0 +1,96 @@
+import numpy as np
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+from regions import CircleSkyRegion
+from synphot import Gaussian1D, SpectralElement
+
+from .. import skygrid
+from ..constraints import (
+    AirmassConstraint,
+    AltitudeConstraint,
+    AtNightConstraint,
+    MoonSeparationConstraint,
+)
+from ..dynamics import Slew
+from ..observer import EarthFixedObserverLocation
+from ..synphot import Detector
+from ..synphot.background import GalacticBackground, ZodiacalBackground
+from ._core import Mission
+
+lsst = Mission(
+    name="lsst",
+    fov=CircleSkyRegion(center=SkyCoord(0 * u.deg, 0 * u.deg), radius=1.75 * u.deg),
+    constraints=[
+        AirmassConstraint(max=2.5, min=1),
+        AltitudeConstraint(min=20 * u.deg, max=85 * u.deg),
+        AtNightConstraint.twilight_civil(),
+        AtNightConstraint.twilight_astronomical(),
+        MoonSeparationConstraint(30 * u.deg),
+    ],
+    detector=Detector(
+        npix=6,  # https://github.com/lsst/rubin_sim/blob/2bf176a6d98ff4c84c352912c5e0721e330fc217/rubin_sim/skybrightness/sky_model.py#L144C19-L144C26
+        plate_scale=(0.2 * u.arcsec) ** 2,
+        # Circular aperture with a diameter of 6.423 m
+        area=np.pi * np.square(0.5 * 6.423 * 100 * u.cm),
+        bandpasses={
+            "sdssu": SpectralElement(
+                Gaussian1D,
+                amplitude=0.25,
+                mean=3600 * u.angstrom,
+                stddev=400 * u.angstrom,
+            ),
+            "ps1__g": SpectralElement(
+                Gaussian1D,
+                amplitude=0.49,
+                mean=4750 * u.angstrom,
+                stddev=750 * u.angstrom,
+            ),
+            "ps1__r": SpectralElement(
+                Gaussian1D,
+                amplitude=0.6,
+                mean=6250 * u.angstrom,
+                stddev=750 * u.angstrom,
+            ),
+            "ps1__i": SpectralElement(
+                Gaussian1D,
+                amplitude=0.69,
+                mean=7550 * u.angstrom,
+                stddev=950 * u.angstrom,
+            ),
+            "ps1__z": SpectralElement(
+                Gaussian1D,
+                amplitude=0.69,
+                mean=8750 * u.angstrom,
+                stddev=750 * u.angstrom,
+            ),
+            "ps1__y": SpectralElement(
+                Gaussian1D,
+                amplitude=0.35,
+                mean=10000 * u.angstrom,
+                stddev=1000 * u.angstrom,
+            ),
```

Comment:
Accurate LSST filter bandpasses are definitely public knowledge, and there is no need to use Gaussian approximations. I don't know what the official reference is for LSST, but here are some general-purpose sources that will at least supply the PanSTARRS and SDSS filter bandpasses:

- https://astroquery.readthedocs.io/en/latest/svo_fps/svo_fps.html
- https://stsynphot.readthedocs.io/en/latest/stsynphot/appendixb_nonhst.html#stsynphot-appendixb-nonhst
- https://github.com/bmorris3/tynt (although this last one may be unmaintained now; @bmorris3, what should we be using instead?)

Reply:
You can now use [`bandpass_from_svo`](https://m4opt.readthedocs.io/en/latest/api/m4opt.synphot.bandpass_from_svo.html#m4opt.synphot.bandpass_from_svo) to retrieve [all LSST transmission curves from SVO](http://svo2.cab.inta-csic.es/svo/theory/fps/index.php?mode=browse&gname=LSST).

Reply:
You can do that - I don't know what version they pulled. 
You can also look at https://github.com/lsst/throughputs